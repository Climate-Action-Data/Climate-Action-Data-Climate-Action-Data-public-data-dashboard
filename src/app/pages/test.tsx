'use client'
import { type NextPage } from "next";

const Test: NextPage = () => {
    return (
        <div>
            <h1>Hello</h1>
            <p>data</p>
        </div>
    )
}

export default Test