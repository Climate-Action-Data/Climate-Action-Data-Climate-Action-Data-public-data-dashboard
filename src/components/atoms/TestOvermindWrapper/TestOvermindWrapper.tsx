import { createOvermind } from 'overmind'
import { Provider } from 'overmind-react'
import { config } from '@/overmind'
import { PropsWithChildren } from 'react'

const overmind = createOvermind(config)

export const TestOvermindWrapper = (props: PropsWithChildren) => {
  return <Provider value={overmind}>{props.children}</Provider>
}
