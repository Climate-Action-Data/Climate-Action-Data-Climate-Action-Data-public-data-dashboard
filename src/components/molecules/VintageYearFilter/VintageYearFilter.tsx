import { Button, Flex, HStack, Popover, PopoverBody, PopoverContent, PopoverTrigger, Spacer, VStack } from '@chakra-ui/react'
import { DropDownIcon } from '@/components/atoms/DropDownIcon/DropDownIcon'
import { FC, useEffect, useState } from 'react'
import { MinusIcon } from '@/components/atoms/MinusIcon/MinusIcon'
import { useTranslation } from 'react-i18next'
import { YearsFilter } from '@/@types/ProjectSearchFilterValues'
import YearInput from '@/components/molecules/YearInput/YearInput'

interface VintageYearFilterProp {
  label: string
  earliestYear?: number
  latestYear?: number
  yearFilter?: YearsFilter
  onYearChange: (payload: YearsFilter) => void
  isResultsPage?: boolean
}

const VintageYearFilter: FC<VintageYearFilterProp> = (prop) => {
  const { label, earliestYear, latestYear, onYearChange, yearFilter, isResultsPage } = prop

  const [minimumYear, setMinimumYear] = useState<number | undefined>()
  const [maximumYear, setMaximumYear] = useState<number | undefined>()

  const { t } = useTranslation(`search`)

  useEffect(() => {
    setMinimumYear(yearFilter?.minYear)
    setMaximumYear(yearFilter?.maxYear)
  }, [yearFilter])

  const handleOnClearClick = () => {
    setMinimumYear(undefined)
    setMaximumYear(undefined)
    onYearChange({})
  }

  const handleOnApplyClick = () => {
    if (minimumYear && maximumYear) {
      if (minimumYear <= maximumYear) {
        onYearChange({ maxYear: maximumYear, minYear: minimumYear })
      }
    } else if (minimumYear) {
      onYearChange({ minYear: minimumYear })
    } else if (maximumYear) {
      onYearChange({ maxYear: maximumYear })
    }
  }

  const handleMinDateOnChange = (year: number | undefined) => {
    setMinimumYear(year)
  }

  const handleMaxDateOnChange = (year: number | undefined) => {
    setMaximumYear(year)
  }

  const renderLabel = () => {
    if (yearFilter?.minYear && yearFilter?.maxYear) {
      return `${yearFilter.minYear} - ${yearFilter.maxYear}`
    }
    if (yearFilter?.minYear) {
      return t(`andLater`, { date: yearFilter.minYear })
    }
    if (yearFilter?.maxYear) {
      return t(`upTo`, { date: yearFilter.maxYear })
    }
    return label
  }

  const generateVariant = () => {
    if (isResultsPage) {
      return yearFilter?.minYear || yearFilter?.maxYear ? `dropdownSelectedDark` : `dropdownUnselectedDark`
    }
    return yearFilter?.minYear || yearFilter?.maxYear ? `dropdownSelected` : `dropdownUnselected`
  }

  return (
    <Popover gutter={0} isLazy placement={`bottom`} matchWidth>
      <PopoverTrigger>
        <Button variant={generateVariant()}>
          <Flex fontWeight={`normal`} fontSize={`16px`} alignItems={`center`} grow={1}>
            {renderLabel()}
            <Spacer />
            <DropDownIcon />
          </Flex>
        </Button>
      </PopoverTrigger>
      <PopoverContent
        zIndex={401}
        border={0}
        borderRadius={`8px`}
        width={[`100%`, null, null, `min-content`]}
        height={`min-content`}
        color={`green.900`}
        backgroundColor={`white`}
        boxShadow={`xl`}
      >
        <PopoverBody width={`100%`}>
          <VStack>
            <HStack width={`100%`} justify={`center`} alignItems={`end`}>
              <YearInput label={`Minimum Date`} value={minimumYear} maxYear={latestYear ?? maximumYear} minYear={earliestYear} onChange={handleMinDateOnChange} />
              <MinusIcon />
              <YearInput label={`Maximum Date`} value={maximumYear} maxYear={latestYear} minYear={earliestYear ?? minimumYear} onChange={handleMaxDateOnChange} />
            </HStack>
            <HStack justify={`space-between`} width={`100%`} margin={`8px`}>
              <Button variant={`textLink`} onClick={handleOnClearClick} data-testid={`vintage-year-filter-clear`}>
                {t(`clear`)}
              </Button>
              <Button variant={`textLink`} onClick={handleOnApplyClick} data-testid={`vintage-year-filter-apply`}>
                {t(`apply`)}
              </Button>
            </HStack>
          </VStack>
        </PopoverBody>
      </PopoverContent>
    </Popover>
  )
}

export default VintageYearFilter
