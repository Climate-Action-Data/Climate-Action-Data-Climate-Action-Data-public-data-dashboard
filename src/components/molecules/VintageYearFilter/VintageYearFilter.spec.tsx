import { fireEvent, render, screen } from '@testing-library/react'
import { TestOvermindWrapper } from '@/test/TestOvermindWrapper'
import userEvent from '@testing-library/user-event'
import VintageYearFilter from '@/components/molecules/VintageYearFilter/VintageYearFilter'

const DEFAULT_DELAY = 500

describe(`VintageYearFilter`, () => {
  const tYear = 2023
  const tLabel = `label`
  const user = userEvent.setup({ delay: null })

  test(`renders a VintageYearFilter and open it`, () => {
    const tOnYearChange = jest.fn()
    render(
      <TestOvermindWrapper>
        <VintageYearFilter label={tLabel} onYearChange={tOnYearChange} />
      </TestOvermindWrapper>,
    )

    fireEvent.click(screen.getByText(tLabel))

    expect(screen.getByRole(`button`, { expanded: true })).toBeTruthy()
  })

  test(`renders a VintageYearFilter and open it and esc`, () => {
    const tOnYearChange = jest.fn()
    render(
      <TestOvermindWrapper>
        <VintageYearFilter label={tLabel} onYearChange={tOnYearChange} />
      </TestOvermindWrapper>,
    )

    fireEvent.click(screen.getByText(tLabel))
    expect(screen.getByRole(`button`, { expanded: true })).toBeTruthy()
    fireEvent.keyDown(screen.getByTestId(`vintage-popover`), {
      key: `Escape`,
      code: `Escape`,
      keyCode: 27,
      charCode: 27,
    })
  })

  const DEFAULT_INPUT_NUMBER = 2

  test(`renders a VintageYearFilter and open it and enter`, () => {
    const tOnYearChange = jest.fn()
    render(
      <TestOvermindWrapper>
        <VintageYearFilter label={tLabel} onYearChange={tOnYearChange} />
      </TestOvermindWrapper>,
    )

    fireEvent.click(screen.getByText(tLabel))
    expect(screen.getByRole(`button`, { expanded: true })).toBeTruthy()
    const inputs = screen.getAllByRole(`textbox`)
    expect(inputs.length).toBe(DEFAULT_INPUT_NUMBER)

    fireEvent.keyDown(inputs[0], {
      key: `Enter`,
      code: `Enter`,
      keyCode: 13,
      charCode: 13,
    })
    fireEvent.keyDown(inputs[1], {
      key: `Enter`,
      code: `Enter`,
      keyCode: 13,
      charCode: 13,
    })
  })

  test(`renders a VintageYearFilter with a pre-selected max year and click on apply`, () => {
    const tOnYearChange = jest.fn()

    const tLabel2 = `Up to ${tYear}`

    render(
      <TestOvermindWrapper>
        <VintageYearFilter label={tLabel} onYearChange={tOnYearChange} yearFilter={{ maxYear: tYear }} />
      </TestOvermindWrapper>,
    )
    fireEvent.click(screen.getByText(tLabel2))
    fireEvent.click(screen.getByTestId(`vintage-year-filter-apply`))
  })

  test(`renders a VintageYearFilter with a pre-selected min year and click on apply`, () => {
    const tOnYearChange = jest.fn()

    const tLabel2 = `${tYear} and later`

    render(
      <TestOvermindWrapper>
        <VintageYearFilter label={tLabel} onYearChange={tOnYearChange} yearFilter={{ minYear: tYear }} />
      </TestOvermindWrapper>,
    )
    fireEvent.click(screen.getByText(tLabel2))
    fireEvent.click(screen.getByTestId(`vintage-year-filter-apply`))
  })

  test(`renders a VintageYearFilter with a pre-selected min year and click on clear`, () => {
    const tOnYearChange = jest.fn()

    const tLabel2 = `${tYear} and later`

    render(
      <TestOvermindWrapper>
        <VintageYearFilter label={tLabel} onYearChange={tOnYearChange} yearFilter={{ minYear: tYear }} />
      </TestOvermindWrapper>,
    )
    fireEvent.click(screen.getByText(tLabel2))
    fireEvent.click(screen.getByTestId(`vintage-year-filter-clear`))

    expect(tOnYearChange).toBeCalledTimes(1)
  })

  test(`renders a VintageYearFilter with a pre-selected min year and max year and click on apply`, () => {
    const tOnYearChange = jest.fn()

    const tLabel2 = `${tYear} - ${tYear}`

    render(
      <TestOvermindWrapper>
        <VintageYearFilter label={tLabel} onYearChange={tOnYearChange} yearFilter={{ minYear: tYear, maxYear: tYear }} />
      </TestOvermindWrapper>,
    )
    fireEvent.click(screen.getByText(tLabel2))
    fireEvent.click(screen.getByTestId(`vintage-year-filter-apply`))
  })

  test(`renders a VintageYearFilter and edit the min year and max year`, async () => {
    const tOnYearChange = jest.fn()

    render(
      <TestOvermindWrapper>
        <VintageYearFilter label={tLabel} onYearChange={tOnYearChange} />
      </TestOvermindWrapper>,
    )

    fireEvent.click(screen.getByText(tLabel))
    const tTextBoxes = screen.getAllByRole(`textbox`)
    await user.type(tTextBoxes[0], `2023`)
    await user.type(tTextBoxes[1], `2023`)
    await new Promise((resolve) => setTimeout(resolve, DEFAULT_DELAY)) // Delay for component update
    fireEvent.click(screen.getByTestId(`vintage-year-filter-apply`))
    await new Promise((resolve) => setTimeout(resolve, DEFAULT_DELAY)) // Delay for component update
    expect(tOnYearChange).toBeCalledTimes(1)
  })
})
