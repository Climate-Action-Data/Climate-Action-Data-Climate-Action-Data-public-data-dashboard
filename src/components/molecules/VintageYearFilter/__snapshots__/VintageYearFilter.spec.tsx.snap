// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VintageYearFilter renders a VintageYearFilter and edit the min year and max year 1`] = `
<div>
  <button
    aria-controls="popover-content-:rb:"
    aria-expanded="true"
    aria-haspopup="dialog"
    class="chakra-button css-4xx2wk"
    id="popover-trigger-:rb:"
    type="button"
  >
    <div
      class="css-jmqhys"
    >
      label
      <div
        class="css-17xejub"
      />
      <svg
        class="chakra-icon css-c0air1"
        focusable="false"
        viewBox="0 0 8 4"
      >
        <path
          d="M3.99911 3.9034L0.191406 0.0957031H7.80681L3.99911 3.9034Z"
          fill="currentColor"
        />
      </svg>
    </div>
  </button>
  <div
    class="chakra-popover__popper css-0"
    data-popper-escaped=""
    data-popper-placement="bottom"
    data-popper-reference-hidden=""
    style="visibility: visible; inset: 0 auto auto 0; --popper-transform-origin: top center; position: absolute; left: 0px; top: 0px; margin: 0px; transform: translate(0px, 0px); width: 0px;"
  >
    <section
      aria-describedby="popover-body-:rb:"
      class="chakra-popover__content css-1j9y0nx"
      id="popover-content-:rb:"
      role="dialog"
      style="transform-origin: var(--popper-transform-origin); opacity: 0.9758993984141853; visibility: visible; transform: scale(0.9987949699207093) translateZ(0);"
      tabindex="-1"
    >
      <div
        class="chakra-popover__body css-8atqhb"
        id="popover-body-:rb:"
      >
        <div
          class="chakra-stack css-tl3ftk"
        >
          <div
            class="chakra-stack css-h0nw9k"
          >
            <div
              class="chakra-stack css-m3jj3s"
            >
              <p
                class="chakra-text css-0"
              >
                Minimum Date
              </p>
              <div
                class="chakra-input__group css-1yb751z"
                data-group="true"
              >
                <input
                  class="chakra-input css-p5mn3g"
                  placeholder="yyyy"
                  value="2023"
                />
              </div>
            </div>
            <svg
              class="chakra-icon css-n059si"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M5.99997 12.75C5.78747 12.75 5.60936 12.678 5.46563 12.5342C5.32188 12.3904 5.25 12.2122 5.25 11.9997C5.25 11.7871 5.32188 11.609 5.46563 11.4654C5.60936 11.3218 5.78747 11.25 5.99997 11.25H18C18.2125 11.25 18.3906 11.3219 18.5344 11.4657C18.6781 11.6095 18.75 11.7877 18.75 12.0003C18.75 12.2129 18.6781 12.391 18.5344 12.5346C18.3906 12.6782 18.2125 12.75 18 12.75H5.99997Z"
                fill="currentColor"
              />
            </svg>
            <div
              class="chakra-stack css-m3jj3s"
            >
              <p
                class="chakra-text css-0"
              >
                Maximum Date
              </p>
              <div
                class="chakra-input__group css-1yb751z"
                data-group="true"
              >
                <input
                  class="chakra-input css-p5mn3g"
                  placeholder="yyyy"
                  value="2023"
                />
              </div>
            </div>
          </div>
          <div
            class="chakra-stack css-1fqndue"
          >
            <button
              class="chakra-button css-4xx2wk"
              data-testid="vintage-year-filter-clear"
              type="button"
            >
              Clear
            </button>
            <button
              class="chakra-button css-4xx2wk"
              data-testid="vintage-year-filter-apply"
              type="button"
            >
              Apply
            </button>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`VintageYearFilter renders a VintageYearFilter and open it 1`] = `
<div>
  <button
    aria-controls="popover-content-:r1:"
    aria-expanded="true"
    aria-haspopup="dialog"
    class="chakra-button css-4xx2wk"
    id="popover-trigger-:r1:"
    type="button"
  >
    <div
      class="css-jmqhys"
    >
      label
      <div
        class="css-17xejub"
      />
      <svg
        class="chakra-icon css-c0air1"
        focusable="false"
        viewBox="0 0 8 4"
      >
        <path
          d="M3.99911 3.9034L0.191406 0.0957031H7.80681L3.99911 3.9034Z"
          fill="currentColor"
        />
      </svg>
    </div>
  </button>
  <div
    class="chakra-popover__popper css-0"
    data-popper-escaped=""
    data-popper-placement="bottom"
    data-popper-reference-hidden=""
    style="visibility: visible; inset: 0 auto auto 0; --popper-transform-origin: top center; position: absolute; left: 0px; top: 0px; margin: 0px; transform: translate(0px, 0px); width: 0px;"
  >
    <section
      aria-describedby="popover-body-:r1:"
      class="chakra-popover__content css-1j9y0nx"
      id="popover-content-:r1:"
      role="dialog"
      style="transform-origin: var(--popper-transform-origin); opacity: 0.3627904654422309; visibility: visible; transform: scale(0.9681395232721115) translateZ(0);"
      tabindex="-1"
    >
      <div
        class="chakra-popover__body css-8atqhb"
        id="popover-body-:r1:"
      >
        <div
          class="chakra-stack css-tl3ftk"
        >
          <div
            class="chakra-stack css-h0nw9k"
          >
            <div
              class="chakra-stack css-m3jj3s"
            >
              <p
                class="chakra-text css-0"
              >
                Minimum Date
              </p>
              <div
                class="chakra-input__group css-1yb751z"
                data-group="true"
              >
                <input
                  class="chakra-input css-p5mn3g"
                  placeholder="yyyy"
                  value=""
                />
              </div>
            </div>
            <svg
              class="chakra-icon css-n059si"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M5.99997 12.75C5.78747 12.75 5.60936 12.678 5.46563 12.5342C5.32188 12.3904 5.25 12.2122 5.25 11.9997C5.25 11.7871 5.32188 11.609 5.46563 11.4654C5.60936 11.3218 5.78747 11.25 5.99997 11.25H18C18.2125 11.25 18.3906 11.3219 18.5344 11.4657C18.6781 11.6095 18.75 11.7877 18.75 12.0003C18.75 12.2129 18.6781 12.391 18.5344 12.5346C18.3906 12.6782 18.2125 12.75 18 12.75H5.99997Z"
                fill="currentColor"
              />
            </svg>
            <div
              class="chakra-stack css-m3jj3s"
            >
              <p
                class="chakra-text css-0"
              >
                Maximum Date
              </p>
              <div
                class="chakra-input__group css-1yb751z"
                data-group="true"
              >
                <input
                  class="chakra-input css-p5mn3g"
                  placeholder="yyyy"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="chakra-stack css-1fqndue"
          >
            <button
              class="chakra-button css-4xx2wk"
              data-testid="vintage-year-filter-clear"
              type="button"
            >
              Clear
            </button>
            <button
              class="chakra-button css-4xx2wk"
              data-testid="vintage-year-filter-apply"
              type="button"
            >
              Apply
            </button>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`VintageYearFilter renders a VintageYearFilter with a pre-selected max year and click on apply 1`] = `
<div>
  <button
    aria-controls="popover-content-:r3:"
    aria-expanded="true"
    aria-haspopup="dialog"
    class="chakra-button css-4xx2wk"
    id="popover-trigger-:r3:"
    type="button"
  >
    <div
      class="css-jmqhys"
    >
      Up to 2023
      <div
        class="css-17xejub"
      />
      <svg
        class="chakra-icon css-c0air1"
        focusable="false"
        viewBox="0 0 8 4"
      >
        <path
          d="M3.99911 3.9034L0.191406 0.0957031H7.80681L3.99911 3.9034Z"
          fill="currentColor"
        />
      </svg>
    </div>
  </button>
  <div
    class="chakra-popover__popper css-0"
    data-popper-escaped=""
    data-popper-placement="bottom"
    data-popper-reference-hidden=""
    style="visibility: visible; inset: 0 auto auto 0; --popper-transform-origin: top center; position: absolute; left: 0px; top: 0px; margin: 0px; transform: translate(0px, 0px); width: 0px;"
  >
    <section
      aria-describedby="popover-body-:r3:"
      class="chakra-popover__content css-1j9y0nx"
      id="popover-content-:r3:"
      role="dialog"
      style="transform-origin: var(--popper-transform-origin); opacity: 0.3340281235578004; visibility: visible; transform: scale(0.96670140617789) translateZ(0);"
      tabindex="-1"
    >
      <div
        class="chakra-popover__body css-8atqhb"
        id="popover-body-:r3:"
      >
        <div
          class="chakra-stack css-tl3ftk"
        >
          <div
            class="chakra-stack css-h0nw9k"
          >
            <div
              class="chakra-stack css-m3jj3s"
            >
              <p
                class="chakra-text css-0"
              >
                Minimum Date
              </p>
              <div
                class="chakra-input__group css-1yb751z"
                data-group="true"
              >
                <input
                  class="chakra-input css-p5mn3g"
                  placeholder="yyyy"
                  value=""
                />
              </div>
            </div>
            <svg
              class="chakra-icon css-n059si"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M5.99997 12.75C5.78747 12.75 5.60936 12.678 5.46563 12.5342C5.32188 12.3904 5.25 12.2122 5.25 11.9997C5.25 11.7871 5.32188 11.609 5.46563 11.4654C5.60936 11.3218 5.78747 11.25 5.99997 11.25H18C18.2125 11.25 18.3906 11.3219 18.5344 11.4657C18.6781 11.6095 18.75 11.7877 18.75 12.0003C18.75 12.2129 18.6781 12.391 18.5344 12.5346C18.3906 12.6782 18.2125 12.75 18 12.75H5.99997Z"
                fill="currentColor"
              />
            </svg>
            <div
              class="chakra-stack css-m3jj3s"
            >
              <p
                class="chakra-text css-0"
              >
                Maximum Date
              </p>
              <div
                class="chakra-input__group css-1yb751z"
                data-group="true"
              >
                <input
                  class="chakra-input css-p5mn3g"
                  placeholder="yyyy"
                  value="2023"
                />
              </div>
            </div>
          </div>
          <div
            class="chakra-stack css-1fqndue"
          >
            <button
              class="chakra-button css-4xx2wk"
              data-testid="vintage-year-filter-clear"
              type="button"
            >
              Clear
            </button>
            <button
              class="chakra-button css-4xx2wk"
              data-testid="vintage-year-filter-apply"
              type="button"
            >
              Apply
            </button>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`VintageYearFilter renders a VintageYearFilter with a pre-selected min year and click on apply 1`] = `
<div>
  <button
    aria-controls="popover-content-:r5:"
    aria-expanded="true"
    aria-haspopup="dialog"
    class="chakra-button css-4xx2wk"
    id="popover-trigger-:r5:"
    type="button"
  >
    <div
      class="css-jmqhys"
    >
      2023 and later
      <div
        class="css-17xejub"
      />
      <svg
        class="chakra-icon css-c0air1"
        focusable="false"
        viewBox="0 0 8 4"
      >
        <path
          d="M3.99911 3.9034L0.191406 0.0957031H7.80681L3.99911 3.9034Z"
          fill="currentColor"
        />
      </svg>
    </div>
  </button>
  <div
    class="chakra-popover__popper css-0"
    data-popper-escaped=""
    data-popper-placement="bottom"
    data-popper-reference-hidden=""
    style="visibility: visible; inset: 0 auto auto 0; --popper-transform-origin: top center; position: absolute; left: 0px; top: 0px; margin: 0px; transform: translate(0px, 0px); width: 0px;"
  >
    <section
      aria-describedby="popover-body-:r5:"
      class="chakra-popover__content css-1j9y0nx"
      id="popover-content-:r5:"
      role="dialog"
      style="transform-origin: var(--popper-transform-origin); opacity: 0.303765981370816; visibility: visible; transform: scale(0.9659405844882712) translateZ(0);"
      tabindex="-1"
    >
      <div
        class="chakra-popover__body css-8atqhb"
        id="popover-body-:r5:"
      >
        <div
          class="chakra-stack css-tl3ftk"
        >
          <div
            class="chakra-stack css-h0nw9k"
          >
            <div
              class="chakra-stack css-m3jj3s"
            >
              <p
                class="chakra-text css-0"
              >
                Minimum Date
              </p>
              <div
                class="chakra-input__group css-1yb751z"
                data-group="true"
              >
                <input
                  class="chakra-input css-p5mn3g"
                  placeholder="yyyy"
                  value="2023"
                />
              </div>
            </div>
            <svg
              class="chakra-icon css-n059si"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M5.99997 12.75C5.78747 12.75 5.60936 12.678 5.46563 12.5342C5.32188 12.3904 5.25 12.2122 5.25 11.9997C5.25 11.7871 5.32188 11.609 5.46563 11.4654C5.60936 11.3218 5.78747 11.25 5.99997 11.25H18C18.2125 11.25 18.3906 11.3219 18.5344 11.4657C18.6781 11.6095 18.75 11.7877 18.75 12.0003C18.75 12.2129 18.6781 12.391 18.5344 12.5346C18.3906 12.6782 18.2125 12.75 18 12.75H5.99997Z"
                fill="currentColor"
              />
            </svg>
            <div
              class="chakra-stack css-m3jj3s"
            >
              <p
                class="chakra-text css-0"
              >
                Maximum Date
              </p>
              <div
                class="chakra-input__group css-1yb751z"
                data-group="true"
              >
                <input
                  class="chakra-input css-p5mn3g"
                  placeholder="yyyy"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="chakra-stack css-1fqndue"
          >
            <button
              class="chakra-button css-4xx2wk"
              data-testid="vintage-year-filter-clear"
              type="button"
            >
              Clear
            </button>
            <button
              class="chakra-button css-4xx2wk"
              data-testid="vintage-year-filter-apply"
              type="button"
            >
              Apply
            </button>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`VintageYearFilter renders a VintageYearFilter with a pre-selected min year and click on clear 1`] = `
<div>
  <button
    aria-controls="popover-content-:r7:"
    aria-expanded="true"
    aria-haspopup="dialog"
    class="chakra-button css-4xx2wk"
    id="popover-trigger-:r7:"
    type="button"
  >
    <div
      class="css-jmqhys"
    >
      2023 and later
      <div
        class="css-17xejub"
      />
      <svg
        class="chakra-icon css-c0air1"
        focusable="false"
        viewBox="0 0 8 4"
      >
        <path
          d="M3.99911 3.9034L0.191406 0.0957031H7.80681L3.99911 3.9034Z"
          fill="currentColor"
        />
      </svg>
    </div>
  </button>
  <div
    class="chakra-popover__popper css-0"
    data-popper-escaped=""
    data-popper-placement="bottom"
    data-popper-reference-hidden=""
    style="visibility: visible; inset: 0 auto auto 0; --popper-transform-origin: top center; position: absolute; left: 0px; top: 0px; margin: 0px; transform: translate(0px, 0px); width: 0px;"
  >
    <section
      aria-describedby="popover-body-:r7:"
      class="chakra-popover__content css-1j9y0nx"
      id="popover-content-:r7:"
      role="dialog"
      style="transform-origin: var(--popper-transform-origin); opacity: 0.6008218588249292; visibility: visible; transform: scale(0.9800410929412464) translateZ(0);"
      tabindex="-1"
    >
      <div
        class="chakra-popover__body css-8atqhb"
        id="popover-body-:r7:"
      >
        <div
          class="chakra-stack css-tl3ftk"
        >
          <div
            class="chakra-stack css-h0nw9k"
          >
            <div
              class="chakra-stack css-m3jj3s"
            >
              <p
                class="chakra-text css-0"
              >
                Minimum Date
              </p>
              <div
                class="chakra-input__group css-1yb751z"
                data-group="true"
              >
                <input
                  class="chakra-input css-p5mn3g"
                  placeholder="yyyy"
                  value=""
                />
              </div>
            </div>
            <svg
              class="chakra-icon css-n059si"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M5.99997 12.75C5.78747 12.75 5.60936 12.678 5.46563 12.5342C5.32188 12.3904 5.25 12.2122 5.25 11.9997C5.25 11.7871 5.32188 11.609 5.46563 11.4654C5.60936 11.3218 5.78747 11.25 5.99997 11.25H18C18.2125 11.25 18.3906 11.3219 18.5344 11.4657C18.6781 11.6095 18.75 11.7877 18.75 12.0003C18.75 12.2129 18.6781 12.391 18.5344 12.5346C18.3906 12.6782 18.2125 12.75 18 12.75H5.99997Z"
                fill="currentColor"
              />
            </svg>
            <div
              class="chakra-stack css-m3jj3s"
            >
              <p
                class="chakra-text css-0"
              >
                Maximum Date
              </p>
              <div
                class="chakra-input__group css-1yb751z"
                data-group="true"
              >
                <input
                  class="chakra-input css-p5mn3g"
                  placeholder="yyyy"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="chakra-stack css-1fqndue"
          >
            <button
              class="chakra-button css-4xx2wk"
              data-testid="vintage-year-filter-clear"
              type="button"
            >
              Clear
            </button>
            <button
              class="chakra-button css-4xx2wk"
              data-testid="vintage-year-filter-apply"
              type="button"
            >
              Apply
            </button>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`VintageYearFilter renders a VintageYearFilter with a pre-selected min year and max year and click on apply 1`] = `
<div>
  <button
    aria-controls="popover-content-:r9:"
    aria-expanded="true"
    aria-haspopup="dialog"
    class="chakra-button css-4xx2wk"
    id="popover-trigger-:r9:"
    type="button"
  >
    <div
      class="css-jmqhys"
    >
      2023 - 2023
      <div
        class="css-17xejub"
      />
      <svg
        class="chakra-icon css-c0air1"
        focusable="false"
        viewBox="0 0 8 4"
      >
        <path
          d="M3.99911 3.9034L0.191406 0.0957031H7.80681L3.99911 3.9034Z"
          fill="currentColor"
        />
      </svg>
    </div>
  </button>
  <div
    class="chakra-popover__popper css-0"
    data-popper-escaped=""
    data-popper-placement="bottom"
    data-popper-reference-hidden=""
    style="visibility: visible; inset: 0 auto auto 0; --popper-transform-origin: top center; position: absolute; left: 0px; top: 0px; margin: 0px; transform: translate(0px, 0px); width: 0px;"
  >
    <section
      aria-describedby="popover-body-:r9:"
      class="chakra-popover__content css-1j9y0nx"
      id="popover-content-:r9:"
      role="dialog"
      style="transform-origin: var(--popper-transform-origin); opacity: 0.28823506491607986; visibility: visible; transform: scale(0.964411753245804) translateZ(0);"
      tabindex="-1"
    >
      <div
        class="chakra-popover__body css-8atqhb"
        id="popover-body-:r9:"
      >
        <div
          class="chakra-stack css-tl3ftk"
        >
          <div
            class="chakra-stack css-h0nw9k"
          >
            <div
              class="chakra-stack css-m3jj3s"
            >
              <p
                class="chakra-text css-0"
              >
                Minimum Date
              </p>
              <div
                class="chakra-input__group css-1yb751z"
                data-group="true"
              >
                <input
                  class="chakra-input css-p5mn3g"
                  placeholder="yyyy"
                  value="2023"
                />
              </div>
            </div>
            <svg
              class="chakra-icon css-n059si"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M5.99997 12.75C5.78747 12.75 5.60936 12.678 5.46563 12.5342C5.32188 12.3904 5.25 12.2122 5.25 11.9997C5.25 11.7871 5.32188 11.609 5.46563 11.4654C5.60936 11.3218 5.78747 11.25 5.99997 11.25H18C18.2125 11.25 18.3906 11.3219 18.5344 11.4657C18.6781 11.6095 18.75 11.7877 18.75 12.0003C18.75 12.2129 18.6781 12.391 18.5344 12.5346C18.3906 12.6782 18.2125 12.75 18 12.75H5.99997Z"
                fill="currentColor"
              />
            </svg>
            <div
              class="chakra-stack css-m3jj3s"
            >
              <p
                class="chakra-text css-0"
              >
                Maximum Date
              </p>
              <div
                class="chakra-input__group css-1yb751z"
                data-group="true"
              >
                <input
                  class="chakra-input css-p5mn3g"
                  placeholder="yyyy"
                  value="2023"
                />
              </div>
            </div>
          </div>
          <div
            class="chakra-stack css-1fqndue"
          >
            <button
              class="chakra-button css-4xx2wk"
              data-testid="vintage-year-filter-clear"
              type="button"
            >
              Clear
            </button>
            <button
              class="chakra-button css-4xx2wk"
              data-testid="vintage-year-filter-apply"
              type="button"
            >
              Apply
            </button>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;
